src/macaron/policy_engine/prelude/helper_rules.dl

```diff
@@ -42,6 +45,12 @@ transitive_dependency(parent, dependency) :-
 .decl is_repo(repo_id: number, repo_complete_name: symbol, component_id: number)
  is_repo(repo_id, repo_complete_name, component_id) :- repository(repo_id, component_id, repo_complete_name, _, _, _, _, _, _, _, _, _, _).
 
+/**
+ *  Extract the id and PURL from the repository relation.
+ */
+.decl is_repo_purl(repo_id: number, purl: symbol)
+ is_repo_purl(repo_id, purl) :- repository(_, repo_id, _, _, _, _, _, purl, _, _, _, _, _).
```

Comment:
We don't store `purl` in the repository` table. We store the remote path, which is the repository URL.

```suggestion
/**
 *  Extract the id and URL from the repository relation.
 */
.decl is_repo_url(repo_id: number, url: symbol)
 is_repo_url(repo_id, url) :- repository(_, repo_id, _, _, _, _, _, url, _, _, _, _, _).
```